### DISCUSSION TITLE
Slow printing strings with color?

### DISCUSSION THREAD

**Discussion by Tetraslam** [Textualize/rich#1213] [Q&A] [answered]
Is it possible to slowly print a string with colors and/or bold/underline/italics? I know you can slow print using the `sys` module:

```
def slowprint(s):
	for c in s + '\n':
		sys.stdout.write(c)
		sys.stdout.flush()
		time.sleep(0.04)
```
However I'd also like to add colors to the text when it gets slow-printed, but right now I'm not able to figure out if there is an in-built function in rich to do both simultaneously on one string. If it doesn't exist, I think it would be a nice feature to include!

**Comment by willmcgugan** [comment, marked as answer]
```python
from time import sleep
from rich.console import Console
from rich.text import Text


text = Text.from_markup("[b]slow[/b] [magenta]printing[/magenta] for [u]Tetraslam")
console = Console()
for c in text:
    console.print(c, end="")
    sleep(0.1)
```

**Reply by Tetraslam** [reply]
Thanks! This works well 😄
